files/acl/restrict-unpublished-files.php

```diff
@@ -38,6 +38,30 @@ function check_file_visibility( $file_visibility, $file_path ) {
 		return FILE_IS_PUBLIC;
 	}
 
+	$attachment_restrict_file_meta = get_post_meta($attachment_id, '_restrict_attachment', true);
+
+	if ( ! empty($attachment_restrict_file_meta)) {
+		
+		if (in_array( $attachment_restrict_file_meta, [true, 'true', 'yes', 1, '1'])) {
```

Comment:
Use `true` as the third argument to `in_array()` (makes it a strict check).

Reply:
I had this as a strict check previously - however, I experienced inconsistent conditional evaluation of the value. 🤔 

I'm going to run through tests again to see if there is some pattern of behavior and report back.

Reply:
No problems in testing. This is good to go.